### ISSUE TITLE
Dispatch in tests?

### ISSUE THREAD

**Issue by kilohsakul** [VeriFIT/mata#341]
I think testing algorithms using dispatch functions is annoying to debug, one needs to spend time figuring out what the was actually called.
Any thoughs? 
It is true though that writing tests is faster.

**Comment by Adda0** [comment]
I disagree. I see no problem with debugging dispatch functions. In tests, you should test both the dispatch function (general test on the corresponding operation) and each specific implementation (testing the correct implementation of the specific algorithm, perf-testing the specific algorithm, correct handling of inputs specific to that algorithm, ...).
